ingeniamotion/communication.py

```diff
@@ -128,133 +109,112 @@ def __servo_connect(
         ip: str,
         dict_path: str,
         alias: str,
-        protocol: Protocol = Protocol.UDP,
         port: int = 1061,
-        reconnection: Optional[dict] = None,
+        connection_timeout: int = 5,
         servo_status_listener: bool = False,
         net_status_listener: bool = False,
     ) -> None:
-        if reconnection is None:
-            reconnection = {}
-        reconnection = {x: reconnection[x] for x in reconnection if reconnection[x] is not None}
         if not path.isfile(dict_path):
-            raise FileNotFoundError("{} file does not exist!".format(dict_path))
+            raise FileNotFoundError(f"{dict_path} file does not exist!")
 
         self.mc.net[alias] = EthernetNetwork()
         net = self.mc.net[alias]
         servo = net.connect_to_slave(
             ip,
             dict_path,
             port,
-            protocol,
-            **reconnection,
+            connection_timeout,
             servo_status_listener=servo_status_listener,
             net_status_listener=net_status_listener,
         )
 
         self.mc.servos[alias] = servo
         self.mc.servo_net[alias] = alias
 
-    def connect_servo_ecat(
+    def connect_servo_eoe_service(
         self,
         ifname: str,
         dict_path: str,
+        ip: str = "192.168.3.22",
         slave: int = 1,
-        eoe_comm: bool = True,
+        port: int = 1061,
         alias: str = DEFAULT_SERVO,
-        reconnection_retries: Optional[int] = None,
-        reconnection_timeout: Optional[int] = None,
         servo_status_listener: bool = False,
         net_status_listener: bool = False,
     ) -> None:
-        """Connect servo by ECAT with embedded master.
+        """Connect to target servo by Ethernet over EtherCAT
 
         Args:
             ifname : interface name. It should have format
                 ``\\Device\\NPF_[...]``.
             dict_path : servo dictionary path.
+            ip : IP address to be assigned to the servo.
             slave : slave index. ``1`` by default.
-            eoe_comm : use eoe communications if ``True``,
-                if ``False`` use SDOs. ``True`` by default.
+            port : servo port. ``1061`` by default.
             alias : servo alias to reference it. ``default`` by default.
-            reconnection_retries : Number of reconnection retried before declaring
-                a connected or disconnected stated.
-            reconnection_timeout : Time in ms of the reconnection timeout.
             servo_status_listener : Toggle the listener of the servo for
                 its status, errors, faults, etc.
             net_status_listener : Toggle the listener of the network
                 status, connection and disconnection.
 
         Raises:
+            TypeError: If the dict_path argument is missing.
```

Comment:
Why did you add this? dict_path is not optional, so of course an error will be raised if the argument is missing. I think it is redundant.

Reply:
My bad, I'll remove it.